subsys/bluetooth/audio/ascs.c

```diff
@@ -932,8 +929,12 @@ static void ase_process(struct k_work *work)
 
 	bt_gatt_notify(ase->ascs->conn, &attr, ase_buf.data, ase_buf.len);
 
-	if (ase->ep.status.state == BT_AUDIO_EP_STATE_RELEASING &&
-	    ase->ep.stream == NULL) {
+	if (ase->ep.status.state == BT_AUDIO_EP_STATE_RELEASING) {
+		if (ase->ep.stream) {
+			ascs_ep_unbind_audio_iso(&ase->ep);
+			bt_audio_stream_detach(ase->ep.stream);
+		}
```

Comment:
If the EP is going into the releasing state here, then `ase->ep.stream` should always be a non-NULL value. We might want to change the `if` to an ASSERT.